ddtrace/internal/writer.py

```diff
@@ -532,6 +536,10 @@ def write(self, spans=None):
                     # are initialized
                     if asbool(os.getenv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", True)):
                         telemetry_writer.enable()
+
+                    if self._report_telemetry_metrics:
```

Comment:
Health metrics are collected and sent in the agent writer while telemetry events should be independent of the agent writer. We enable telemetry collection in the agent writer so that the first telemetry event is sent when the the first trace is sent, but this can change in the future. Ideally we should have one global flag that determines whether telemetry collection is enabled, having two might cause confusion and unexpected behavior down the line. I’d prefer to remove this argument and field from the AgentWriter but if this makes testing significantly more challenging I’m fine with keeping this code as is.